From 5d281572d0afbf69d934737e1ee4c553670a46cc Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Thu, 29 May 2014 19:22:27 +0200
Subject: [PATCH 4/5] auto/lib/libxslt/conf: allow to override ngx_feature_path
 and ngx_feature_libs

Because libxml2 headers are not in /usr/include by default, hardcoding the
include directory to /usr/include/libxml2 does not play well when
cross-compiling, or if libxml2 has been installed somewhere else.

This patch allows to define/override the libxslt include directory, and
the libxslt libs flags.

Being able to override the include location is especially useful when
cross-compiling to prevent gcc from complaining about unsafe include
location for cross-compilation (-Wpoision-system-directories).

So far, this warning is only triggered by libxslt.

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
---
 auto/lib/libxslt/conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/auto/lib/libxslt/conf b/auto/lib/libxslt/conf
index bc19d83..386f1a0 100644
--- a/auto/lib/libxslt/conf
+++ b/auto/lib/libxslt/conf
@@ -12,8 +12,8 @@
                       #include <libxslt/xsltInternals.h>
                       #include <libxslt/transform.h>
                       #include <libxslt/xsltutils.h>"
-    ngx_feature_path="/usr/include/libxml2"
-    ngx_feature_libs="-lxml2 -lxslt"
+    ngx_feature_path="${ngx_feature_path_libxslt:=/usr/include/libxml2}"
+    ngx_feature_libs="${ngx_feature_libs_libxslt:=-lxml2 -lxslt}"
     ngx_feature_test="xmlParserCtxtPtr    ctxt = NULL;
                       xsltStylesheetPtr   sheet = NULL;
                       xmlDocPtr           doc;
--
1.9.2

